File: necform/src/main/java/org/sid/necform/service/InscriptionService.java
package org.sid.necform.service;

import org.sid.necform.dto.request.CreateInscriptionRequest;
import org.sid.necform.dto.request.UpdateInscriptionRequest;
import org.sid.necform.dto.response.InscriptionResponse;

import java.util.List;
import java.util.UUID;

public interface InscriptionService {

    InscriptionResponse create(CreateInscriptionRequest request);

    List<InscriptionResponse> getAll();

    InscriptionResponse getById(UUID id);

    InscriptionResponse update(UUID id, UpdateInscriptionRequest request);

    void delete(UUID id);
}

File: necform-front/src/app/core/services/inscription.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { 
  Inscription, 
  CreateInscription, 
  UpdateInscription 
} from '../../shared/models/inscription.model';
import { ApiService } from './api.service';

@Injectable({
  providedIn: 'root'
})
export class InscriptionService {
  private readonly endpoint: string;

  constructor(
    private http: HttpClient,
    private apiService: ApiService
  ) {
    this.endpoint = `${this.apiService.getApiUrl()}/inscriptions`;
  }

  create(request: CreateInscription): Observable<Inscription> {
    return this.http.post<Inscription>(this.endpoint, request);
  }

  getAll(): Observable<Inscription[]> {
    return this.http.get<Inscription[]>(this.endpoint);
  }

  getById(id: string): Observable<Inscription> {
    return this.http.get<Inscription>(`${this.endpoint}/${id}`);
  }

  update(id: string, request: UpdateInscription): Observable<Inscription> {
    return this.http.put<Inscription>(`${this.endpoint}/${id}`, request);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.endpoint}/${id}`);
  }
}
